From b8e4b587bc5fbadaa270dfa76f134f0b9a3537f8 Mon Sep 17 00:00:00 2001
From: Stefan Wilke <stefan.wilke@asideas.de>
Date: Tue, 18 Feb 2020 17:19:14 +0100
Subject: [PATCH] fix fat jar bundling

---
 build.gradle.kts | 25 +++++++------------------
 1 file changed, 7 insertions(+), 18 deletions(-)

diff --git a/build.gradle.kts b/build.gradle.kts
index 18caafe..27da0f5 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -5,7 +5,7 @@
  * For more details take a look at the Java Quickstart chapter in the Gradle
  * User Manual available at https://docs.gradle.org/6.1.1/userguide/tutorial_java_projects.html
  */
-import org.gradle.jvm.tasks.Jar
+
 plugins {
     // Apply the java plugin to add support for Java
     java
@@ -48,22 +48,11 @@ tasks.withType<Jar> {
     manifest {
         attributes["Main-Class"] = "src.main.java.crawler.Crawler"
     }
-}
 
-
-val fatJar = task("fatJar", type = Jar::class) {
-    baseName = "${project.name}-fat"
-    // manifest Main-Class attribute is optional.
-    // (Used only to provide default main class for executable jar)
-    manifest {
-        attributes["Main-Class"] = "src.main.java.crawler.Crawler"
-    }
-    from(configurations.runtime.get().map({ if (it.isDirectory) it else zipTree(it) }))
-    with(tasks["jar"] as CopySpec)
+	from(sourceSets.main.get().output)
+    dependsOn(configurations.runtimeClasspath)
+    from({
+    	configurations.runtimeClasspath.get().filter {
+    		it.name.endsWith("jar") }.map { zipTree(it) }
+	})
 }
-
-tasks {
-    "build" {
-        dependsOn(fatJar)
-    }
-}
\ No newline at end of file
-- 
2.25.0

